src/util/graph.h

```diff
@@ -60,6 +61,36 @@ class graph_nodet
   {
     out.erase(n);
   }
+
+  /// \brief Node with attributes suitable for Graphviz DOT format
+  ///
+  /// Derived types may override this function to produce more informative DOT
+  /// diagrams than the default implementation, which displays only the node
+  /// index. The return value should either be an empty string, or a single line
+  /// representing this node that can be parsed by `dot`. This line would
+  /// typically be the node index, followed by whitespace, followed by a list of
+  /// node attributes within square brackets. Here is a sample implementation
+  /// for a fictional node type with `is_evil()` and `is_pink()` functions:
+  ///
+  ///     std::stringstream ss;
+  ///     ss << std::to_string(idx)
+  ///        << " [shape=\"" << is_evil() ? "box" : "diamond"
+  ///        << "\", color=\"" << is_pink() ? "#e91e63" : "#9c27b0"
+  ///        << "\", label=\"this is node " << std::to_string(idx)
+  ///        << "\"]";
+  ///     return ss.str();
+  ///
+  /// If this function returns an empty string for a particular node,
+  /// grapht::output_dot() method will not output this node at all, nor any of
+  /// its incoming or outgoing edges.
```

Comment:
Maybe use an optional type where the empty optional triggers this behavior. I'm not sure the actual behavior is exactly as describe, wouldn't we still output some edges which are coming into to these nodes?

Reply:
OK, I'm going to remove this from this PR and add a cleaner implementation in a different PR